fasta2JSON.R

```diff
@@ -0,0 +1,128 @@
+head(All_ncovid_NoNGS_data,10)
+dim(All_ncovid_NoNGS_data)
+# Generating a LIMS for this run
+# ========================
+LIMS<-rbind(NGS_covid_data,All_ncovid_NoNGS_data)
+head(LIMS, 20)
+colnames(LIMS)<-c("sampleNumber","lastName", "firstName", "DOB","collectionDate")
+head(M)
+PatientDemographic<-merge(M,LIMS,by.x = "Sample_Accession", by.y ="sampleNumber", all.x = T)
+
+head(PatientDemographic,10)
+# (3) Getting data from epitrax_export_1551
+cat(crayon::blue$underline("Please wait, serching epitrax_export_155","\n"))
+# ========================
+EpiTrax_Export<-rownames(fileSnapshot("/Volumes/IDGenomics_NAS/COVID/daily_metadata/epitrax/",
+                                      pattern=glob2rx("export_1551*.csv"))$info[which.max(fileSnapshot("/Volumes/IDGenomics_NAS/COVID/daily_metadata/epitrax/",
+                                                                                                       pattern=glob2rx("export_1551*.csv"))$info$mtime),])
+cat(crayon::blue$underline("Please wait, searching EpiTrax","\n")); head(EpiTrax_Export)
+locus<-paste('/Volumes/IDGenomics_NAS/COVID/daily_metadata/epitrax/',EpiTrax_Export,sep="")
+M_epitrax<-fread(locus, select =c("patient_record_number","lab_accession_no"))
+M_epitrax<-cbind(M_epitrax,M_epitrax$lab_accession_no); colnames(M_epitrax)[3]<-"Epitrax_ID"
+head(M_epitrax)
+Epi<-unique(merge(M,M_epitrax, by.x ="Sample_Accession", by.y = "lab_accession_no", all.x = T))
+head(Epi,20)
+Epi$patient_record_number<-as.character(Epi$patient_record_number)
+Epi$patient_record_number[is.na(Epi$patient_record_number)] <- "missing"
+Epi$Epitrax_ID[is.na(Epi$Epitrax_ID)] <- "not_found"
+# all together
+# ======================
+Matrix<- cbind(Epi,PatientDemographic)
+Matrix$Epitrax_ID[is.na(Matrix$Epitrax_ID)] <- "Not_found"
+head(Matrix,30)
+
+#Getting the sequences: only PatientSamples
+# ======================
+cat(crayon::blue$underline("Please wait, getting the fasta sequences","\n"))
+path<-paste(runPath,"covidseq_output/Sample_Analysis/",sep="/")
+
+
+head(M,21)
+seq_name<-0; sequence<-0
+counter = length(M$Sample_Accession); counter
+
+for (i in 1:counter){
+  sample=Matrix$Sample_Accession[i]; sample
+  Lyapunovv<- Sys.glob(paste(path,sample,"*/*.fasta",sep = ""),dirmark = TRUE); Lyapunovv
+  fastaFile <- readDNAStringSet(Lyapunovv)
+  fastaFile
+  seq_name[i] = names(fastaFile)
+  sequence[i] = paste(fastaFile)
+}
+df <- data.frame(seq_name, sequence);
+# JSON file
+# ===================
+Alist_1 = vector(mode="list", length = counter)
+Alist_1[[1]] = Matrix$Sample_Accession[1:counter]
+Alist_1[[2]] = Matrix$collectionDate[1:counter]
+Alist_1[[3]] = Matrix$patient_record_number[1:counter]
+Alist_1[[4]] = Matrix$Epitrax_ID[1:counter]
+Alist_1[[5]] = df
+
+
+json2 <- data.frame(
+  LIMS_ID = Alist_1[[1]],
+  collection_date = Alist_1[[2]],
+  Epitrax_ID = Alist_1[[3]],
+  filename = Alist_1[[4]],
+  sequence = df)
```

Comment:
`sequence = df` prints strangely to the resultant file

Reply:
It as been modify to sprint the results as required